### ISSUE TITLE
Support finer-grained control over hugepages allocations

### ISSUE THREAD

**Issue by jalaziz** [aws/aws-nitro-enclaves-cli#602]
The current behavior of the `nitro-enclaves-allocator` script is to allocate the largest possible page first, then move on to smaller pages as necessary. While this generally makes sense, it can result in over-allocation of memory when you want to run multiple enclaves on a single node.

For example, if I want to allocate 2GB of memory for huge pages, the allocator will allocate 2 1GB pages. However, now if I'm using Kubernetes and want to schedule one pod that uses 1.5GB of memory and another of 500MB, I am unable to unless I request more memory than needed.

It would be wonderful if the max page size could be controlled via the `allocator.yaml` config. Even better would be fine-grained control over how many of each page size to allocate.

**Comment by matteobjornsson** [comment]
Agreed. It's a bit unexpected when using the cli. It would be great to specify more granular control!

For example, if I allocate 1024 MiB for the entire instance, with the expectation that am going to start a few enclaves with 256 MiB memory, when I run an enclave
```shell
$ nitro-cli run-enclave --cpu-count 2 --memory 256 --eif-path enclave.eif
```
 It hands over the entire 1024 page:
```shell
Start allocating memory...
Started enclave with enclave-cid: 16, memory: 1024 MiB, cpu-ids: [1, 5]
{
  "EnclaveName": "enclave",
  "EnclaveID": "i-0edc49d9de17846a9-enc198aa328094d2bb",
  "ProcessID": 4551,
  "EnclaveCID": 16,
  "NumberOfCPUs": 2,
  "CPUIDs": [
    1,
    5
  ],
  "MemoryMiB": 1024
}
```
Preventing the ability to start another enclave with 256 MiB
```shell
$ nitro-cli run-enclave --cpu-count 2 --memory 256 --eif-path enclave.eif
Start allocating memory...
[ E27 ] Insufficient memory available. User provided `memory` is 256 MB, which is more than the available hugepage memory.
```

**Comment by alirezasalamat** [comment]
No update for this problem?  :(
